worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/StorageManager.scala

```diff
@@ -558,6 +560,10 @@ final private[worker] class StorageManager(conf: CelebornConf, workerSource: Abs
         if (removedFileInfos != null) {
           removedFileInfos.asScala.foreach {
             case (_, fileInfo) =>
+              if (memCacheManager.contains(fileInfo.getFilePath)) {
```

Comment:
Multiple threads can simultaneously call `cleanupExpiredShuffleKey`. IMO we can change `removeCache` as follows:
```
  public void removeCache(String key) {
    ByteBuf cache = caches.remove(key);
    if (cache != null) {
      currentCacheSize.getAndAdd(-1 * cache.readableBytes());
      cache.release();
    }
  }
```
Here we can just call `removeCache` without checking contains.